queries/fortran/highlights.scm

```diff
@@ -117,69 +272,66 @@
   ">="
   "=="
   "/="
+  "//"
+  (assumed_rank)
 ] @operator
 
 [
   "\\.and\\."
   "\\.or\\."
+  "\\.eqv\\."
+  "\\.neqv\\."
   "\\.lt\\."
   "\\.gt\\."
-  "\\.ge\\."
   "\\.le\\."
+  "\\.ge\\."
   "\\.eq\\."
-  "\\.eqv\\."
-  "\\.neqv\\."
+  "\\.ne\\."
+  "\\.not\\."
 ] @keyword.operator
 
- ;; Brackets
- [
-  "("
-  ")"
-  "["
-  "]"
-  "<<<"
-  ">>>"
- ] @punctuation.bracket
-
- ;; Delimiter
- [
-  "::"
+; Punctuation
+
+[ "[" "]" ] @punctuation.bracket
+
+[ "(" ")" ] @punctuation.bracket
+
+[ "<<<" ">>>" ] @punctuation.bracket
+
+(array_literal
+  ["(/" "/)"] @punctuation.bracket)
+
+[
+  ":"
   ","
+  "/"
   "%"
- ] @punctuation.delimiter
-
-(parameters
-  (identifier) @parameter)
+  "::"
+  "=>"
+] @punctuation.delimiter
 
-(program_statement
-  (name) @namespace)
+; Literals
 
-(module_statement
-  (name) @namespace)
+(string_literal) @string
 
-(function_statement
-  (name) @function)
+(assumed_size) @character.special
+(data_value "*" @character.special)
+(format_identifier "*" @character.special)
+(unit_identifier "*" @character.special)
```

Comment:
`@character` is for char literals

Reply:
those all are just asterisks which are special

Reply:
But they are not character literals. CharSpecial is another of those confusing categories. I could not think of a single accusation where this could be needed.

Reply:
Think `\n` or `\t`.

Reply:
Okay I'll just leave it under `@operator` then